### PR TITLE
chore: clean up obvious eslint warnings, mostly unused variables

### PR DESCRIPTION
**Pull request by gasp**
reduced eslint warnings from 

✖ 360 problems (2 errors, 358 warnings)

to

✖ 271 problems (0 errors, 271 warnings)

fixing only the obvious ones with no business impact

### CODE REVIEW

**Comment by mwithi** [comment]
very nice work @gasp, please let's wait @SteveGT96 review. Thanks!

**Comment by SilverD3** [comment]
A lot of eslint warning has been removed, almost all. But there are still some warning related to scss files.

**Comment by gasp** [comment]
Thanks for your code review !
As I am not yet very familiarized with the codebase, I avoid making any change that could impact the business logic of the app. Let's say that this is a _first pass_ till I gain more confidence in making changes on this app ^^

**Comment by SilverD3** [comment]
The app is building.

File: src/components/accessories/billTable/types.ts [hunk 1]
```diff
@@ -30,3 +30,14 @@ export type BillFilterFormFieldName =
   | "status"
   | "month"
   | "year";
+
```
[the remark this answers is not in the document]
Comment (mwithi) [reply src/components/accessories/billTable/types.ts:33]: @gasp do you need advice on this point? 
Cc: @SilverD3

**Comment by gasp** [comment]
Yes, it's building and it works great !
This is a tremendous job and a great app, I was just trying to reduce some noise in the build phase as I was reading the app's code to get more familiar with it !

If you think that this commit does not add enough value, adds complexity or is not the proper way to address it, I am totally fine with closing (and not merging this PR ☺️)

File: src/components/accessories/billTable/types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (gasp) [reply src/components/accessories/billTable/types.ts:33]: If there is a better way to do this, please let me know.
You can also use the github ["suggest change" feature](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) or contact me on slack !

**Comment by mwithi** [comment]
I definetely love this kind of "chores" 😍
Let's resolve the conversation above (there is a PR check for it) and we can merge for me.
Thanks!!

File: src/components/accessories/billTable/types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mwithi) [reply src/components/accessories/billTable/types.ts:33]: do you refer to `TBillDisplayData` ?

**Comment by gasp** [comment]
"chore" is used in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/#specification) to non-production changes, but it's more a convention than a specification, and [after further readings](https://medium.com/@callmeryan/semantic-commit-messages-bcd60f75de1f), these should be more *style* than *chore*

**Comment by mwithi** [comment]
> "chore" is used in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/#specification) to non-production changes, but it's more a convention than a specification, and [after further readings](https://medium.com/@callmeryan/semantic-commit-messages-bcd60f75de1f), these should be more _style_ than _chore_

Very useful reference! Thanks!

Moreover, on this project we always try to stick to Jira's issues (see [here](https://openhospital.atlassian.net/jira/software/c/projects/OP/issues)) and link them to the code on GitHub with the issues code either in the commit message or the branch name, but at least in the PR title.

For other small, self-explaining and not "release-note relevant" changes, we can use the convention above.
In short, "chore" was totally fine for me in this case ;)

**Review by SteveGT96 — APPROVED** [review]
Changes look OK for me.

**Review by mwithi — APPROVED** [review]
Approved by @SteveGT96 and @SilverD3. Thanks a lot @gasp!
